src/Compilers/CSharp/Portable/CodeGen/EmitAddress.cs

```diff
@@ -339,7 +345,7 @@ private LocalSymbol DigForValueLocal(BoundSequence topSequence, BoundExpression
         /// Checks if expression directly or indirectly represents a value with its own home. In
         /// such cases it is possible to get a reference without loading into a temporary.
         /// </summary>
-        private bool HasHome(BoundExpression expression, bool needWriteable)
+        private bool HasHome(BoundExpression expression, AddressKind addressKind)
```

Comment:
`addressKind` here refers to the destination, so maybe call it `destinationAddressKind` or `requiredAddressKind`?

Reply:
I would like to rename this "HasHome" to something like "CanGetDirectReference" or similar (in a separate clean-up change). I started doing that, but reverted since that does not need to be in 15.5 and just makes the PR larger than it needs to be.
Then "requiredAddressKind" would be Ok.

---
In reply to: [146960213](https://github.com/dotnet/roslyn/pull/22772#discussion_r146960213) [](ancestors = 146960213)